src/MySqlConnector/Core/ServerSession.cs

```diff
@@ -596,6 +596,63 @@ public async Task DisposeAsync(IOBehavior ioBehavior, CancellationToken cancella
 		return statusInfo;
 	}
 
+	public async Task PostConnectionsQueriesAsync(ConnectionSettings cs,
+		IOBehavior ioBehavior, CancellationToken cancellationToken)
+	{
+		var payloadDatas = new List<PayloadData>();
+		var responses = new List<Func<Task>>();
+		var okReponse = async () =>
+		{
+			var payload = await ReceiveReplyAsync(ioBehavior, cancellationToken).ConfigureAwait(false);
+			OkPayload.Create(payload.Span, SupportsDeprecateEof, SupportsSessionTrack);
+		};
+
+		// set 'collation_connection' to the server default
+		payloadDatas.Add(m_setNamesPayload);
+		responses.Add(okReponse);
+
+		// if server permit it, set default timeout for commands
+		if (cs.DefaultCommandTimeout > 0 && SupportPerQueryVariables)
+		{
+			payloadDatas.Add(QueryPayload.Create(false,
+				Encoding.ASCII.GetBytes("SET SESSION max_statement_time=" + cs.DefaultCommandTimeout)));
+			responses.Add(okReponse);
+		}
+
+		// retrieve real server version
+		if (ShouldGetRealServerDetails(cs))
+		{
+			payloadDatas.Add(SupportsQueryAttributes ? s_selectConnectionIdVersionWithAttributesPayload : s_selectConnectionIdVersionNoAttributesPayload);
+			responses.Add(async () =>
+			{
+				await ReadRealServerDetailsAsync(ioBehavior, cancellationToken).ConfigureAwait(false);
+			});
+		}
+
+		var querySendTasks = new List<Task>();
+		if (m_supportsPipelining)
+		{
+			foreach (var payloadData in payloadDatas)
+			{
+				querySendTasks.Add(SendAsync(payloadData, ioBehavior, cancellationToken).AsTask());
```

Comment:
I don't know if my code makes the guarantee that `SendAsync` can be called in parallel (e.g., if a payload ends up getting broken up across multiple `Socket` calls). This may be unsafe.